serializer.rst

```diff
@@ -2,223 +2,443 @@ How to Use the Serializer
 =========================
 
 Symfony provides a serializer to serialize/deserialize to and from objects and
-different formats (e.g. JSON or XML). Before using it, read the
-:doc:`Serializer component docs </components/serializer>` to get familiar with
-its philosophy and the normalizers and encoders terminology.
+different formats (e.g. JSON or XML).
 
 .. _activating_the_serializer:
 
 Installation
 ------------
 
 In applications using :ref:`Symfony Flex <symfony-flex>`, run this command to
-install the ``serializer`` :ref:`Symfony pack <symfony-packs>` before using it:
+install the serializer :ref:`Symfony pack <symfony-packs>` before using it:
 
 .. code-block:: terminal
 
     $ composer require symfony/serializer-pack
 
-Using the Serializer Service
-----------------------------
+.. note::
 
-Once enabled, the serializer service can be injected in any service where
-you need it or it can be used in a controller::
+    The serializer pack also installs some commonly used optional
+    dependencies of the Serializer component. When using this component
+    outside the Symfony framework, you might want to start with the
+    ``symfony/serializer`` package and install optional dependencies if you
+    need them.
 
-    // src/Controller/DefaultController.php
-    namespace App\Controller;
+.. seealso::
 
-    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
-    use Symfony\Component\Serializer\SerializerInterface;
+    A popular alternative to the Symfony Serializer component is the third-party
```

Comment:
Jus asking: is it still OK to mention and promote this alternative serializer? I think that a few years ago the alternative was similar or even better than Symfony's. Is that still true today?

If we keep the recommendation, I'd remove the details about its license because it's not relevant enough.

Reply:
I'm honestly not sure. I moved this directive from the bottom of the page to top, as that is the better place to mention it if we want.

Personally, I've worked mostly in projects using the JMS serializer and the project is still actively maintained (and [used](https://packagist.org/packages/jms/serializer/stats)). I left it in mostly to help a bit when a new developer ends up on this page when joining a Symfony dev team (I can imagine they'll search for serializer in the Symfony docs, not realizing the project actually uses a different serializer and this directive might give them the push they need).